packages/atomic/src/components/search/facets/atomic-automatic-facet-builder/atomic-automatic-facet-builder.tsx

```diff
@@ -0,0 +1,96 @@
+import {BooleanValue, NumberValue, Schema} from '@coveo/bueno';
+import {
+  AutomaticFacetBuilder,
+  AutomaticFacetBuilderState,
+  SearchStatus,
+  buildAutomaticFacetBuilder,
+  buildSearchStatus,
+} from '@coveo/headless';
+import {Component, Prop, State, h} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+/**
+ * @beta - This component is part of the automatic facets feature.
+ * Automatic facets are currently in beta testing and should be available soon.
+ *
+ * The `atomic-automatic-facet-builder` is a component that renders the facets from
+ * the automatic facets feature. It dynamically creates multiple atomic-automatic-facet
+ * components based on the desiredCount prop. Automatic facets are generated by the index
+ * in response to the search query.
+ */
+@Component({
+  tag: 'atomic-automatic-facet-builder',
+  shadow: false,
+})
+export class AtomicAutomaticFacetBuilder implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @State() public error!: Error;
+
+  @BindStateToController('automaticFacetBuilder')
+  @State()
+  private automaticFacetBuilderState!: AutomaticFacetBuilderState;
+
+  public automaticFacetBuilder!: AutomaticFacetBuilder;
+  public searchStatus!: SearchStatus;
+
+  /**
+   * @beta - This prop is part of the automatic facets feature.
+   * Automatic facets are currently in beta testing and should be available soon.
+   *
+   * The desired count of automatic facets.
+   * Must be a positive integer.
+   */
+  @Prop() public desiredCount!: number;
+
+  /**
+   * @beta - This prop is part of the automatic facets feature.
+   * Automatic facets are currently in beta testing and should be available soon.
+   *
+   * Specifies whether the automatic facets are collapsed.
+   */
+  @Prop({reflect: true, mutable: true}) public areCollapsed = true;
```

Comment:
Also, this pr adds a new prop to the component called areCollapsed to control wether or not the individual automatic facets are collapsed by default or not. I added it to help with the sorting of the facet manager so that the index.html looked good. Eventually, I want to better support the automatic facets components with the facet manager feature `collapseFacetsAfter`

Reply:
Just to confirm, this prop only effects the _initial_ state of the facets? So if this prop is `true`, all child facets will be rendered as collapsed at first, but can be individually expanded after the fact?

Reply:
I'm still quite new to Stencil, but I don't **_think_** you need either of the `reflect` nor `mutable` options for this prop as I don't believe it is ever reassigned/changed at render-time 🤔 

Also more variable name nitpicking, I think `collapse` or `collapsed` is better. It's purpose is still clear to the dev without there being the clunkiness of it only being grammatically correct in certain contexts (e.g., if only a single automatic facet is returned by the index, it's not right to say that the facet "are collapsed")

Reply:
Yes that is correct, I can remove those, I missed it thank you.